emperor/core.py

```diff
@@ -128,9 +128,16 @@ class Emperor(object):
     def __init__(self, ordination, mapping_file, dimensions=5):
         self.ordination = ordination
 
+        self.mf = mapping_file.copy()
+
+        # the index of the dataframe needs to be named
+        if self.mf.index.name is None:
+            self.mf.index.name = 'SampleID'
```

Comment:
:-1: Because this will still thrown an error if the index.name doesn't correspond to 'SampleID'.

I think it is better to just remove this line and throw and reasonable error.